stdlib/public/runtime/Casting.cpp

```diff
@@ -2755,6 +2805,165 @@ swift::swift_conformsToProtocol(const Metadata *type,
   goto recur;
 }
 
+static const Metadata *
+_metadataForNode(const ConformanceState &C, const NodePointer node);
+
+// returns true if the generic pattern matches the template type name;
+// note that this will only work for public symbols, but it is better
+// than nothing (or perhaps it isn't).
+static bool
+_genericPatternMatchesTemplate(const void *genericPattern,
+                               llvm::StringRef templateName) {
+  Dl_info info;
+
+  if (!dladdr(genericPattern, &info) || info.dli_sname == NULL)
```

Comment:
I couldn't find a way to get the pattern name from the conformance table (even the existing `ProtocolConformanceRecord::dump()` uses `dladdr()`). Can you give me a pointer on where to find it?